### DISCUSSION TITLE
How are websockets used?

### DISCUSSION THREAD

**Discussion by nvonpentz** [wevm/wagmi#924] [Q&A]
From the [docs](https://wagmi.sh/docs/client#websocketprovider-optional):
> If you provide a WebSocket provider, it will be used instead of polling in certain instances.

Would someone please clarify which instances it's used?  Thanks.

**Comment by tmm** [comment]
Anywhere there is `watch` behavior (e.g. waiting for a transaction to mine, updating on new blocks, etc.)
